File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        HashMap poketmon = new HashMap<>();
        
        for (int temp: nums){
            poketmon.put(temp, true);
        }
        // System.out.println(poketmon.toString());
        // System.out.println(poketmon.size());
        if(poketmon.size()>nums.length / 2){
            return nums.length/2;
        }else{
            return poketmon.size();
        }
        // return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    let answer = 0;
    const map = new Map();
    
    for(let n of nums){
        if(map.get(n)){
            continue;
        }else{
            map.set(n, true)
            answer++;
        }
    }
    
    return answer>(nums.length/2) ? nums.length/2 : answer;
}